presto-spi/src/main/java/com/facebook/presto/spi/security/GrantInfo.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.security;
+
+import com.facebook.presto.spi.SchemaTableName;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+public class GrantInfo
+{
+    private final Set<PrivilegeInfo> privilegeInfo;
+    private final Identity grantee;
+    private final SchemaTableName schemaTableName;
+    private final Optional<Identity> grantor;
+    private final Optional<Boolean> withHierarchy;
```

Comment:
This denotes whether the privilege is automatically granted for all the derived objects too (e.g., on views derived from the table). Hive doesn't seem to keep this information, so I can't retrieve it. (also, I just noticed that there is an optional `WITH HIERARCHY OPTION` clause in `GRANT` statement as per the SQL standard, but Presto doesn't support it yet).